src/command_modules/azure-cli-lab/azure/cli/command_modules/lab/validators.py

```diff
@@ -0,0 +1,225 @@
+    if not namespace.subnet:
+        # Get the first subnet of the lab's virtual network
+        lab_vnet = vnet_operation.get_resource(namespace.resource_group, namespace.lab_name, namespace.vnet_name)
+        namespace.subnet = lab_vnet.subnet_overrides[0].lab_subnet_name
+
+        # Determine value for disallow_public_ip_address based on subnet's use_public_ip_address_permission property
+        if lab_vnet.subnet_overrides[0].use_public_ip_address_permission == 'Allow':
+            namespace.disallow_public_ip_address = 'false'
+        else:
+            namespace.disallow_public_ip_address = 'true'
+
+
+def validate_image_argument(namespace):
+    image_type = _parse_image_argument(namespace)
+    if image_type == 'gallery_image':
+        namespace.gallery_image_reference = GalleryImageReference(offer=namespace.os_offer,
+                                                                  publisher=namespace.os_publisher,
+                                                                  os_type=namespace.os_type,
+                                                                  sku=namespace.os_sku,
+                                                                  version=namespace.os_version)
+        namespace.notes = namespace.notes or namespace.image
+    else:
+        namespace.custom_image_id = namespace.image
+
+
+def _parse_image_argument(namespace):
+    # 1 - check if a fully-qualified ID (assumes it is an image ID)
+    if is_valid_resource_id(namespace.image):
+        return 'image_id'
+
+    # 2 - check if an existing lab Gallery Image Reference
+    try:
+        gallery_image_operation = devtestlabs_management_client.gallery_image
+        odata_filter = "name eq '{}'".format(namespace.image)
+        lab_images = list(gallery_image_operation.list(namespace.resource_group, namespace.lab_name, odata_filter))
+
+        if not lab_images:
+            err = "Unable to find image name '{}' in the '{}' lab Gallery.".format(namespace.image, namespace.lab_name)
+            raise CLIError(err)
+        elif len(lab_images) > 1:
+            err = "Found more than 1 images with name "'{}'". Please pick one from {}"
+            raise CLIError(err.format(namespace.image, [x['name'] for x in lab_images]))
+        else:
+            image = lab_images[0]
+            namespace.os_offer = image.image_reference.offer
+            namespace.os_publisher = image.image_reference.publisher
+            namespace.os_sku = image.image_reference.sku
+            namespace.os_type = image.image_reference.os_type
+            namespace.os_version = image.image_reference.version
+        return 'gallery_image'
+    except CloudError:
+        err = 'Invalid image "{}". Use a custom image id or pick one from lab Gallery'
+        raise CLIError(err.format(namespace.image))
+
+
+# TODO: Following methods can be extracted into common utils shared by other command modules
```

Comment:
I would argue these shouldn't be promoted to core utils. They are only used for VM/VMSS create.